tests/cases/fourslash/jsxTagLinkedEdit1.ts

```diff
@@ -0,0 +1,53 @@
+/// <reference path='fourslash.ts' />
+
+// the content of basic.tsx
+//const jsx = (
+//   <div>
+//   </div>
+//);
+
+// @Filename: /basic.tsx
+////const jsx = (
+////    </*0*/d/*1*/iv/*2*/>/*7*/
+////    </*6*///*3*/di/*4*/v/*5*/>/*8*/
+////);
+
+// @Filename: /whitespaceInvalidClosing.tsx
+////const jsx = (
+////   <div>
+////   < /*9*/ /div>
+////);
+
+// @Filename: /whitespaceOpening.tsx
+////const jsx3 = (
+////   </*10*/ /*11*/div/*12*/ /*13*/> /*14*/
+////   </di/*A*/v>
+////);
+
+// @Filename: /whitespaceClosing.tsx
+////const jsx = (
+////   <di/*B*/v>
+////   <//*15*/ /*16*/div/*17*/ /*18*/> /*19*/
+////);
+
+const linkedCursors1 = {ranges: [{start: 19, length: 3}, 

```

Comment:
The goal of markers/ranges in fourslash is to avoid hardcoded file positions like this in testcases, since they're very tedious to maintain.